### PR TITLE
Token overview (SOFTWARE-5077)

### PR DESCRIPTION
**Pull request by matyasselmeci**

### CODE REVIEW

**Comment by matyasselmeci** [comment]
I still need to write some more background information on OAuth2 and OIDC... and also I haven't actually previewed this in mkdocs yet... but I think it's ready for a first review.

File: docs/security/tokens/overview.md [hunk 1]
```diff
@@ -1,22 +1,186 @@
 DateReviewed: 2022-03-10
-
+=======
```
Comment (brianhlin) [inline docs/security/tokens/overview.md:2]:
```suggestion
```

I don't think this is actually necessary looking at our other docs with `DateReviewed` metadata

File: docs/security/tokens/overview.md [hunk 2]
```diff
@@ -1,22 +1,186 @@
...
+and [WLCG Tokens](https://github.com/WLCG-AuthZ-WG/common-jwt-profile/blob/master/profile.md).
+These standards describe the claims that are used in the payload of the JWT.
+
+SciTokens and WLCG Tokens are similar standards and have some common claims:
+
+**Issuer ("iss")**
+
+The issuer identifies the organization that issued the token.
+An issuer looks like an HTTPS URL;
+this URL must be valid and publicly accessible because it is used by services to validate the token.
```
Comment (brianhlin) [inline docs/security/tokens/overview.md:39]:
```suggestion
this URL must be valid and publicly accessible as they are used by site services to validate the token.
```

File: docs/security/tokens/overview.md [hunk 3]
```diff
@@ -1,22 +1,186 @@
...
+Token issuers will be described below.
```
Comment (brianhlin) [inline docs/security/tokens/overview.md:40]:
```suggestion
Token issuers will be described below.
```

We should add a link to the relevant subsection here

File: docs/security/tokens/overview.md [hunk 4]
```diff
@@ -1,22 +1,186 @@
...
+
+**Subject ("sub")**
+
+The subject identifies an entity (which could be a human or a robot) that owns the token.
+Unlike the subject of an X.509 certificate, a token subject does not need to be globally unique,
+only unique to the issuer.
+Subjects will be elaborated on below.
```
Comment (brianhlin) [inline docs/security/tokens/overview.md:47]:
```suggestion
Subjects will be elaborated on below.
```

Another link

File: docs/security/tokens/overview.md [hunk 5]
```diff
@@ -1,22 +1,186 @@
...
+For example a token used for submitting a job to a CE would have
+`<CE FQDN>:<CE PORT>` in the `aud` claim.
+The special values `ANY` (SciTokens) or `https://wlcg.cern.ch/jwt/v1/any` (WLCG Tokens) allow the token to be
+used on any server.
+
+**Scope ("scope")**
+
+The scope limits the actions that can be made using the token.
+The format of the scope claim differs between SciTokens and WLCG Tokens;
+scopes in use by OSG services will be listed below.
```
Comment (brianhlin) [inline docs/security/tokens/overview.md:66]: Link!

File: docs/security/tokens/overview.md [hunk 6]
```diff
@@ -1,22 +1,186 @@
...
+In addition to generating and signing tokens, token issuers provide a public endpoint that can be used to validate an
+issued token,
+e.g. an OSG Compute Entrypoint (CE) will contact the token issuer to authorize a bearer token used for pilot job
+submission.
+
+The issuer is listed in the `iss` claim; this should be an HTTPS URL of a web server.
+This server must have the public key that can be used to validate the token in a well-known location,
+as described by the [OpenID Connect Discovery standard](https://openid.net/specs/openid-connect-discovery-1_0.html).
+If the issuer is down, or the the public key cannot be downloaded, the token cannot be verified
+and will be rejected.
```
Comment (brianhlin) [inline docs/security/tokens/overview.md:81]: One thing to note is that clients of a token issuer will cache the public key for a few hours (and technically that cache is 4 days right now to help ease the transition to tokens while collabs spin up their token issuers)

File: docs/security/tokens/overview.md [hunk 7]
```diff
@@ -76,3 +240,48 @@ The following collaborations support support file transfer using WebDAV or XRoot
...
+
+-   Online: paste the token into <https://jwt.io>.
+
+-   Offline:
+    1.  Write the token to a file named `tok`.
+    2.  Run `IFS=. read header payload signature < tok`.
+    3.  Run `echo $header | base64 -d` to examine the header.
+        Run `echo $payload | base64 -d` to examine the payload.
+        Note: the header or payload may be missing the final padding characters (up to 2 `=` characters);
+        adding them (e.g. `echo $payload== | base64 -d`) should make base64 stop complaining about "invalid input" or "truncated input".
```
Comment (brianhlin) [inline docs/security/tokens/overview.md:269]: Let's throw this into the client doc instead that's ready

**Review by brianhlin — COMMENTED** [review]
Looks very thorough! I think we've got what we need for now and can add the background stuff later. See additional comments.

File: docs/security/tokens/overview.md (hunk 7 above)
Comment (matyasselmeci) [reply docs/security/tokens/overview.md]: Done

File: docs/security/tokens/overview.md [hunk 8]
```diff
@@ -1,22 +1,186 @@
...
+for further information.
+
+
+Using Bearer Tokens with XRootD
+-------------------------------
+
+In order to support Token AAI, your XRootD installation must be based on OSG 3.6 or OSG 3.5-upcoming.
+You will need XRootD 5.0.2 or newer, with the `xrootd-scitokens` plugin.
+Follow the [configuring XRootD authorization documentation](https://opensciencegrid.org/docs/data/xrootd/xrootd-authorization)
+for information on how to configure XRootD to accept bearer tokens.
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/security/tokens/overview.md]: Moved them to #946

File: docs/security/tokens/overview.md (hunk 6 above)
Comment (matyasselmeci) [reply docs/security/tokens/overview.md:81]: Added

**Review by brianhlin — APPROVED** [review]
LGTM, thanks!

**Pull request merged**